imports/validators.py

```diff
@@ -23,22 +29,61 @@ def __call__(self, value):
             )
 
         with zipfile.ZipFile(value.file) as zfile:
-            required_extensions = ('.shp', '.shx', '.dbf')
-            filenames, extensions = [], []
-            for filename in zfile.namelist():
-                if filename.endswith(required_extensions):
-                    filename, extension = os.path.splitext(filename)
-                    filenames.append(filename)
-                    extensions.append(extension)
-
-            if len(set(filenames)) > 1:
-                raise ValidationError(
-                    self.messages['mismatched_filenames'],
-                    code='mismatched_filenames',
-                )
-
-            if set(extensions) != set(required_extensions):
-                raise ValidationError(
-                    self.messages['missing_required_files'],
-                    code='missing_required_files',
-                )
+            self._validate_files(zfile)
+
+            dbf, shp, shx = sorted([name for name in zfile.namelist() if name.endswith(REQUIRED_SHAPEFILE_EXTENSIONS)])
```

Comment:
This seems to be a redundant file extension check, the same thing is happening in `_validate_files()`. Also, this would not be a good thing to do anyways since there is no guarantee that three names would be present in the list which would lead to an unpack error --> 500